➕ Add Typescript

saschahaselhuhn opened the pull request:
- Adding typescript
- renaming js files to tsx / ts

https://nextjs.org/docs/basic-features/typescript

saschahaselhuhn committed:
➕ Add Typescript

phlgr reviewed (changes requested):
Where is your installed module in the package.json and the package-lock.json?

mewiemers reviewed (approved):
only one file is seen here

M-Sieger reviewed (approved):
gj !

saschahaselhuhn committed:
➕ Installed typescript with types

m-wiemers commented:
It seems something is missing 🌎

saschahaselhuhn committed:
🚚  Renamed JS to TS / TSX files

lmachens commented on package.json:
```diff
@@ -9,9 +9,12 @@
   },
   "dependencies": {
+    "@types/node": "^14.14.33",
+    "@types/react": "^17.0.3",
     "next": "10.0.8",
     "react": "17.0.1",
-    "react-dom": "17.0.1"
+    "react-dom": "17.0.1",
+    "typescript": "^4.2.3"
```
Looks like these should be `devDependencies` and not `dependencies`

saschahaselhuhn committed:
➕ move typescript to dev dependencies

lmachens reviewed (approved):
🚀

phlgr reviewed (approved):
🚀

saschahaselhuhn commented:
- Created tsconfig.json with: touch tsconfig.json
- Ran next with: npm run dev
- Installed required packages with: npm install --save-dev typescript @types/react @types/node
- Renamed _app.js and index.js to _app.tsx and index.tsx
- Renamed hello.js to hello.ts
- Configured tsconfig.json with: npm run dev
- Restarted server with npm run dev

https://nextjs.org/docs/basic-features/typescript
